intellihide.js

```diff
@@ -221,39 +200,36 @@ const Intellihide = new Lang.Class({
 
             let monitorIndex = this._settings.get_int('preferred-monitor');
 
-            if (monitorIndex < 0 || monitorIndex > Main.layoutManager.monitors.length)
+            if ((monitorIndex < 0) || (monitorIndex > Main.layoutManager.monitors.length))
                 monitorIndex = Main.layoutManager.primaryIndex;
 
             let topWindow = null;
-            for (let i = windows.length-1; i>=0; i--) {
+            for (let i = windows.length - 1; i >= 0; i--) {
                 let meta_win = windows[i].get_meta_window();
-                if (this._handledWindow(meta_win)
-                    && meta_win.get_monitor() == monitorIndex) {
+                if (this._handledWindow(meta_win) && (meta_win.get_monitor() == monitorIndex)) {
                     topWindow = meta_win;
                     break;
                 }
             }
 
             if (topWindow !== null) {
-
                 this._topApp = this._tracker.get_window_app(topWindow);
                 // If there isn't a focused app, use that of the window on top
                 this._focusApp = this._tracker.focus_app || this._topApp
 
                 windows = windows.filter(this._intellihideFilterInteresting, this);
 
-                for(let i=0; i< windows.length; i++) {
-
+                for (let i in windows.length) {
```

Comment:
Correct, unintended! Oops, I really did not intend to change any code. Probably it was a bad copy/paste. I apologize for wasting your time trying to find the bug. :/